lib/ts/InkErrno.cc

```diff
@@ -0,0 +1,106 @@
+/** @file
+
+  Error code defines
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#include "InkErrno.h"
+#include <string.h>
+
+const char *
+InkStrerror(int ink_errno)
+{
+  if (ink_errno < INK_START_ERRNO) {
+    return strerror(ink_errno);
```

Comment:
``strerror`` is not threadsafe in glibc, however when you give it a legit ``errno`` it returns a pointer into ``sys_errlist``, which means it works in the 99% case. If you pass junk it will format into a shared buffer.